napari/layers/base/base.py

```diff
@@ -905,15 +912,51 @@ def help(self, help_text):
 
     @property
     def interactive(self):
-        """bool: Determine if canvas pan/zoom interactivity is enabled."""
-        return self._interactive
+        warnings.warn(
+            '`Layer.interactive` is deprecated since 0.5.0 and will be removed in 0.6.0.',
+            category=DeprecationWarning,
+        )
+        return self.mouse_pan or self.mouse_zoom
 
     @interactive.setter
     def interactive(self, interactive):
-        if interactive == self._interactive:
+        warnings.warn(
+            '`Layer.interactive` is deprecated since 0.5.0 and will be removed in 0.6.0.',
+            category=DeprecationWarning,
+        )
+        self.mouse_pan = interactive
+        self.mouse_zoom = interactive
```

Comment:
In the comment https://github.com/napari/napari/pull/5701/files#r1163642973 you point error here. 

As interactive is emitted on set `mouse_pan` and not emitted on set `mouse_zoom` and events in napari are emitted immediately, and `interactive` value is calculated using `or`, then incorrect value of interactive will be present during event emission. 

The proper solution will be to use an event emission blocker for `interactive` during the assignment `mouse_pan` and add interactive emission in `mouse_zoom` assignment. 

```suggestion
        with self.events.interactive.blocker()
            self.mouse_pan = interactive
        self.mouse_zoom = interactive
```

https://github.com/napari/napari/blob/5fa6716fceaa0d77516d664373c3f3a84ccd790d/napari/utils/events/event.py#L867